bottlecap/src/logs/processor.rs

```diff
@@ -200,9 +287,12 @@ impl Processor {
     }
 
     pub fn process(&mut self, event: TelemetryEvent, aggregator: &Arc<Mutex<Aggregator>>) {
-        if let Ok(log) = self.make_log(event) {
+        if let Ok(mut log) = self.make_log(event) {
             let mut aggregator = aggregator.lock().expect("lock poisoned");
-            aggregator.add(log);
+            let should_send_log = self.apply_rules(&mut log);
```

Comment:
ah I didn't realize that log processing rules encompassed both redaction _and_ deciding if we should keep or drop a log.

That API makes sense after some consideration, but it was jarring to me at first

Reply:
Yup, I'm not happy with the readability tbh, I'm open for suggestions! 😿

Reply:
Nah I think it's okay, ship it!